### DISCUSSION TITLE
Add support for inverted RX and TX, and pull-up for RX

### DISCUSSION THREAD

**Discussion by ellepdesk** [emsesp/EMS-ESP32#1421] [Ideas]
Hi, 

I've built an alternate circuit to interface with the EMS bus using a minimum of components:
![image](https://github.com/emsesp/EMS-ESP32/assets/6548222/c439655d-a1a1-4005-a382-ab146e1d3d62)

The only drawback is that the RX and TX are inverted, and RX relies on the internal pullup if the ESP.

I've made a branch to hard-code these changes for me, but it may be useful for other users as well.
[InvertRxTx-pullupRx](https://github.com/emsesp/EMS-ESP32/commit/ee06e7becb9f3ae0fd4fe069d5ba12fd99b06e90)

I would like to expose these settings in the webui, but I have no idea where to start.

**Comment by proddy** [comment]
I'd be ok to merge this and use DEFINES to determine if Rx and Tx are invested - that you can add to the build process (default is off obviously).

**Reply by ellepdesk** [reply]
Cool, I'll rework it to use DEFINES and submit a pull-request
